TESTS/host_tests/usb_device_serial.py

```diff
@@ -64,6 +64,16 @@ def usb_serial_name(serial_number):
     return None
 
 
+if sys.version_info < (3,):
```

Comment:
Can this be done with https://pypi.org/project/six/

Reply:
@madchutney thanks for pointing this out. 👍 I haven't used this module before.

[`six.ensure_str()`](https://six.readthedocs.io/#six.ensure_str) looks exactly like what I need here. Too bad we currently use an older version of this module. https://github.com/ARMmbed/mbed-os/blob/daafb950d96e2e72027f72f1dbdfdd6333cf7fbc/requirements.txt#L8

`six.ensure_str()` was added in v1.12.0 (https://github.com/benjaminp/six/commit/db3d0d678b422614d5de0f9ed76a2112a28c7b19). @madchutney, @0xc0170 can we ask someone to bump the `six` version?

Reply:
@OPpuolitaival, either inside or outside of the function definition; doesn't make any difference for me. I think I left it this way after seeing http://python3porting.com/problems.html#nicer-solutions.

Reply:
No big matter for me

Reply:
> six.ensure_str() was added in v1.12.0 (benjaminp/six@db3d0d6). @madchutney, @0xc0170 can we ask someone to bump the six version?

I've just noticed this question here. We can do it if there is a need, with full testing of course.